test/FxERC20.js

```diff
@@ -0,0 +1,209 @@
+        rootToken = await BridgedToken.deploy("Bridged token", "BERC20");
+        await rootToken.deployed();
+
+        // ERC20 tunnels
+        const FxERC20RootTunnel = await ethers.getContractFactory("FxERC20RootTunnel");
+        fxERC20RootTunnel = await FxERC20RootTunnel.deploy(deployer.address, deployer.address, childToken.address,
+            rootToken.address);
+        await fxERC20RootTunnel.deployed();
+
+        const FxERC20ChildTunnel = await ethers.getContractFactory("FxERC20ChildTunnel");
+        fxERC20ChildTunnel = await FxERC20ChildTunnel.deploy(deployer.address, childToken.address, rootToken.address);
+        await fxERC20ChildTunnel.deployed();
+
+        // Set child and root tunnels accordingly
+        await fxERC20RootTunnel.setFxChildTunnel(fxERC20ChildTunnel.address);
+        await fxERC20ChildTunnel.setFxRootTunnel(fxERC20RootTunnel.address);
+
+        // Mint tokens
+        await childToken.mint(deployer.address, initMint);
+    });
+
+    context("Initialization", async function () {
+        it("Deploying with zero addresses", async function () {
+            const FxERC20RootTunnel = await ethers.getContractFactory("FxERC20RootTunnel");
+            await expect(
+                FxERC20RootTunnel.deploy(AddressZero, AddressZero, AddressZero, AddressZero)
+            ).to.be.revertedWithCustomError(fxERC20RootTunnel, "ZeroAddress");
+
+            await expect(
+                FxERC20RootTunnel.deploy(signers[1].address, AddressZero, AddressZero, AddressZero)
+            ).to.be.revertedWithCustomError(fxERC20RootTunnel, "ZeroAddress");
+
+            await expect(
+                FxERC20RootTunnel.deploy(signers[1].address, signers[1].address, AddressZero, AddressZero)
+            ).to.be.revertedWithCustomError(fxERC20RootTunnel, "ZeroAddress");
+
+            await expect(
+                FxERC20RootTunnel.deploy(signers[1].address, signers[1].address, signers[1].address, AddressZero)
+            ).to.be.revertedWithCustomError(fxERC20RootTunnel, "ZeroAddress");
+
+            const FxERC20ChildTunnel = await ethers.getContractFactory("FxERC20ChildTunnel");
+            await expect(
+                FxERC20ChildTunnel.deploy(AddressZero, AddressZero, AddressZero)
+            ).to.be.revertedWithCustomError(fxERC20ChildTunnel, "ZeroAddress");
+
+            await expect(
+                FxERC20ChildTunnel.deploy(signers[1].address, AddressZero, AddressZero)
+            ).to.be.revertedWithCustomError(fxERC20ChildTunnel, "ZeroAddress");
+
+            await expect(
+                FxERC20ChildTunnel.deploy(signers[1].address, signers[1].address, AddressZero)
+            ).to.be.revertedWithCustomError(fxERC20ChildTunnel, "ZeroAddress");
+        });
+    });
+
+    context("Deposit and withdraw ERC20 tokens", async function () {
+        it("Should fail when trying to call from incorrect contract addresses", async function () {
+            // FxChild as a sender is incorrect
```

Comment:
```suggestion
            // Deployer as a sender is incorrect
```

Reply:
fxERC20RootTunnel is the correct one

Reply:
Here it's the `FxChild`, which is `deployer.address` in this test setup, and not the `signers[1].address.` Corrected.